### PR TITLE
ISPN-3076 Upgrade JBoss Logging 3.2.1

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-3076

### CODE REVIEW

File: core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java [hunk 1]
```diff
@@ -291,8 +291,8 @@ protected final Object handleNonTxWriteCommand(InvocationContext ctx, DataWriteC
       // TotalOrderStateTransferInterceptor doesn't set the topology id for PFERs.
       // TODO Shouldn't PFERs be executed in a tx with total order?
       boolean topologyChanged = isSync && currentTopologyId != commandTopologyId && commandTopologyId != -1;
-      log.tracef("Command topology id is %d, current topology id is %d, successful? %s",
-            commandTopologyId, currentTopologyId, command.isSuccessful());
+      log.tracef("Command topology id is %s, current topology id is %d, successful? %s",
```
Comment (anistor) [inline core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java]: ```%d``` is no longer supported by jboss logging 3.2.1 ?

Comment (tristantarrant) [inline core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java]: Unfortunately jboss logging 3.2 has introduced a bunch of overloaded methods like debugf(String format, int param1, Object param2, Object param3) which are ambiguous with debugf(String format, Object... params);
I'm merely disambiguating, but if you have a better idea, I'm open :)

Comment (anistor) [inline core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java]: The compiler should not have problems disambiguating those. Isn’t this an eclipse or intellij issue actually?

Comment (tristantarrant) [inline core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java]: No, this happens even with plain old javac (1.8.0_45 here)

**Comment by anistor** [comment]
Weird. I've reverted those and compiled with 1.7.0_80 and it works. Why would jboss logging 3.2.1 try to make our life harder?

**Comment by anistor** [comment]
And yes, with JDK 8 I get those errors :) Freaking java 8 !!! :)

**Comment by tristantarrant** [comment]
I think it's Java 8's overeager boxing ?

**Comment by Sanne** [comment]
To keep in mind: http://lists.jboss.org/pipermail/wildfly-dev/2015-February/003668.html

**Comment by anistor** [comment]
Good point @Sanne! I think just adding a dummy (apparently useless) cast to Object for the first arg wherever we have that ambiguity will avoid calls to the newly added methods and will also keep both java 7 and 8 compilers happy. Adding that silly cast is a bit better than dropping all those %d and manually stringizing ints and longs.

**Comment by Sanne** [comment]
The reason for JBoss Logging to have added those new overloaded methods is to avoid autoboxing, as it's a performance cost.
So.. yes you could force it into explicit boxing but then make sure you guard the statement with appropriate "isTraceEnabled()" checks.

**Comment by tristantarrant** [comment]
I'm not really fussy about the performance impact of autoboxing here: these are all debugf and tracef, so rare in production. I'll add the missing if (trace) where needed.

**Comment by Sanne** [comment]
sure, that's why I'm suggesting to add the 'if' statement. Otherwise the autoboxing will happen even if trace is disabled.

**Comment by tristantarrant** [comment]
Yes, but in some cases the logs are already wrapped in conditions (such as
checking for errors) so adding the additional "trace" conditions saves us
nothing and complicates the code

**Comment by tristantarrant** [comment]
Here is an example of the ambiguity: 
BaseDistributionInterceptor.java:[295,13] reference to tracef is ambiguous
[ERROR] both method tracef(java.lang.String,long,long,java.lang.Object) in org.jboss.logging.BasicLogger and method tracef(java.lang.String,long,java.lang.Object,java.lang.Object) in org.jboss.logging.BasicLogger match

the invocation is

```java
log.tracef("Command topology id is %d, current topology id is %d, successful? %s",
               commandTopologyId, currentTopologyId, command.isSuccessful());
```

note that it is choosing:
```java
void tracef(String format, long arg1, long arg2, Object arg3);
```

instead of the more appropriate

```java
void tracef(String format, int arg1, int arg2, Object arg3);
```

casting the first parameter to (Object) works around the issue:

```java
log.tracef("Command topology id is %d, current topology id is %d, successful? %s",
               (Object)commandTopologyId, currentTopologyId, command.isSuccessful());
```

**Comment by anistor** [comment]
IMHO the jboss logging api improvement is a bad thing. Let's cast where needed and move on.

**Comment by tristantarrant** [comment]
Made the changes

**Comment by anistor** [comment]
Pulling ...

**Comment by anistor** [comment]
Integrated. Thanks @tristantarrant !

**Pull request closed without merging**